BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -2310,7 +2310,8 @@ private bool FireLaser()
 
             float timeGap = GetTimeGap();
             beamDuration = Math.Min(timeGap * 0.8f, 0.1f);
-            if ((!pulseLaser || ((timeSinceFired > timeGap) && pulseLaser))
+
+            if (timeSinceFired > timeGap
```

Comment:
Won't removing the `!pulseLaser ||` condition mean that non-pulse lasers (beam lasers?) will then act like pulse lasers? (The ` && pulseLaser` was unnecesary in the other branch of the condition though.)

Alternatively, if `GetTimeGap()` returns `0` for beam lasers, then this would be fine. `timeFired` should probably just be set to `Time.time` instead of the `for` loop on lines 2365—2366 too.

Reply:
Beam lasers are already set to fire once per fixedUpdateTick (RoF 3000 in Start()), but that would be a sensible precaution